lib/ts/fetch.ts

```diff
@@ -567,167 +566,203 @@ export async function onInvalidClaimResponse(response: ResponseWithBody) {
+    // the session is being removed. So we set it to "remove" in the
+    // cookie. This way, when we query for this token, we will return
+    // undefined (see getLocalSessionState), and not refresh the session
+    // unnecessarily.
+    logDebugMessage(`setToken: saved ${tokenType} token into cookies`);
+    return storeInCookies(name, value, expiry);
+}
+
+function storeInCookies(name: string, value: string, expiry: number) {
+    let expires = "Fri, 31 Dec 9999 23:59:59 GMT";
+    if (value !== "remove" && expiry !== Number.MAX_SAFE_INTEGER) {
+        // we must always respect this expiry and not set it to infinite
+        // cause this ties into the session's lifetime. If we set this
+        // to infinite, then a session may not exist, and this will exist,
+        // then for example, if we check a session exists, and this says yes,
+        // then if we getAccessTokenPayload, that will attempt a session refresh which will fail.
+        // Another reason to respect this is that if we don't, then signOut will
+        // call the API which will return 200 (no 401 cause the API thinks no session exists),
+        // in which case, we will not end up firing the SIGN_OUT on handle event.
+        expires = new Date(expiry).toUTCString();
+    }
+    const domain = AuthHttpRequest.config.sessionTokenFrontendDomain;
+    if (
+        domain === "localhost" ||
+        domain === WindowHandlerReference.getReferenceOrThrow().windowHandler.location.getHostName()
+    ) {
+        // since some browsers ignore cookies with domain set to localhost
+        // see https://github.com/supertokens/supertokens-website/issues/25
+        return CookieHandlerReference.getReferenceOrThrow().cookieHandler.setCookie(
+            `${name}=${value};expires=${expires};path=/;samesite=${
+                AuthHttpRequest.config.isInIframe ? "none;secure" : "lax"
+            }`
+        );
+    } else {
+        return CookieHandlerReference.getReferenceOrThrow().cookieHandler.setCookie(
+            `${name}=${value};expires=${expires};domain=${domain};path=/;samesite=${
+                AuthHttpRequest.config.isInIframe ? "none;secure" : "lax"
+            }`
+        );
+    }
+}
+
+export async function getTokenForHeaderAuth(tokenType: TokenType) {
+    const name = getStorageNameForToken(tokenType);
+
+    return getFromCookies(name);
+}
+
+async function getFromCookies(name: string) {
+    let value = "; " + (await CookieHandlerReference.getReferenceOrThrow().cookieHandler.getCookie());
+    let parts = value.split("; " + name + "=");
+    if (parts.length >= 2) {
+        let last = parts.pop();
+        if (last !== undefined) {
+            return last.split(";").shift();
         }
     }
+    return undefined;
+}
 
-    const { status } = await getIdRefreshToken(false);
+async function setTokenHeadersIfRequired(clonedHeaders: Headers, addRefreshToken: boolean = false) {
+    const transferMethod = AuthHttpRequest.config.tokenTransferMethod;
```

Comment:
in this function, we do not check if `transferMethod` is = `header` before setting the authorization header - that is intentional right? If so, please comment this here linking to the relevant ADR(s)

Reply:
It is intentional: the frontend preference might be to use header or cookie-based auth, but that will only take effect after a new session is created, and by default, we can use the existing one until the user logs out.
I'll add this as a comment.